src/bundle-manifest.ts

```diff
@@ -100,11 +100,12 @@ export function generateBundles(depsIndex: TransitiveDependenciesMap):
     Bundle[] {
   const bundles: Bundle[] = [];
 
-  // TODO(usergenic): Assert a valid transitive dependencies map; i.e.
-  // entrypoints should include themselves as dependencies and entrypoints
-  // should *probably* not include other entrypoints as dependencies.
+  const prunedDepsIndex: TransitiveDependenciesMap =
+      pruneTransitiveDependenciesMap(depsIndex);
```

Comment:
I wonder if this should be optional. In some cases it makes sense to merge entrypoints together. I'm worried that if the depsIndex passed to the bundle heuristics is always pruned of entrypoints and their transitive dependencies, it will be much more challenging to perform these optimizations